### PR TITLE
add popup window

### PR DESCRIPTION
**Pull request by Hamzaoutdoors**
- Added CSS and JS events for projects popups
- Added projects object to store data
- Fixed Linters errors

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
++ `<p>${projects[projectId].description}</p>`
++ '<div class="d-flex pop-buttons">'
++ '<button class="btn-primary btn-pop d-flex" type="button">'
++ 'See Live <i class="fa fa-clock-o"></i></button>'
++ '<button class="btn-primary btn-pop d-flex" type="button">'
++ 'See Source <i class="fa fa-github"></i></button>'
++ '</div>'
++ '</div>'
++ '</div>';
+console.log(project[0]) */
```
Comment (carlos-ssh) [inline js/app.js:152]: - Take your project at the next level. Remove unused commented lines.

File: js/app.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    description:
+      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque porro aliquam quod repellat harum distinctio quas voluptates eligendi maiores at facilis asperiores, voluptatem placeat cum nostrum nisi quae, natus neque? Minus expedita, aliquid repellat recusandae laboriosam natus culpa odit.',
+    liveLink: '#',
+    sourceLink: '#',
+  },
+];
+
+const divProject = document.createElement('div');
+
+// add modal function
```
Comment (carlos-ssh) [inline js/app.js:94]: - Kindly remove commented lines.

File: README.md [hunk 3]
```diff
@@ -83,13 +86,6 @@ Give a ⭐️ if you like this project!
 
 ## Acknowledgments
 
-- Module 1 : project portfolio (mobile menu)
-
-## Accessibility Report 
+- Module 1 : project portfolio (popup window)
```
Comment (carlos-ssh) [inline README.md:91]: - Provide a License.

File: js/app.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
+const hamburger = document.querySelector('.header .nav-bar .nav-list .hamburger');
+const menu = document.querySelector('.header .nav-bar .nav-list ul');
+const menuItem = document.querySelectorAll('.header .nav-bar .nav-list ul li a');
+const closeIcon1 = document.querySelector('.menu-icon1');
+const closeIcon2 = document.querySelector('.menu-icon2');
+const openPopupButtons = document.querySelectorAll('.popup-btn');
+
+// navbar menu
```
Comment (carlos-ssh) [inline js/app.js:8]: - Remove unused commented lines.

File: README.md [hunk 5]
```diff
@@ -1,9 +1,10 @@
 # Portfolio project
```
Comment (carlos-ssh) [inline README.md:1]: - Add into your .gitignore file your external files like: .DS_Store, .vscode.

File: js/app.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+// add modal function
+function addEvents() {
+  for (let i = 0; i < openPopupButtons.length; i += 1) {
+    openPopupButtons[i].addEventListener('click', (ev) => {
+      const projectId = ev.target.id;
+      divProject.classList.add('popup-overlay-objects');
+      document.querySelector('.popup-bg').style.display = 'flex';
+      // create techs
+      let techsHTML = '';
+      for (let k = 0; k < projects[i].technologies.length; k += 1) {
```
Comment (carlos-ssh) [inline js/app.js:103]: - Kindly fix this error in chrome console: `` Cannot read property 'technologies' of undefined ``

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @Hamzaoutdoors , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/app.js (hunk 4 above)
Comment (Hamzaoutdoors) [reply js/app.js:8]: Done thanks for your review

File: README.md (hunk 3 above)
Comment (Hamzaoutdoors) [reply README.md:91]: Done thanks for your review

File: README.md (hunk 5 above)
Comment (Hamzaoutdoors) [reply README.md:1]: I don't understand what did you mean by external files

File: js/app.js (hunk 6 above)
Comment (Hamzaoutdoors) [reply js/app.js:103]: Done thanks for your review

File: js/app.js (hunk 2 above)
Comment (Hamzaoutdoors) [reply js/app.js:94]: Done thanks for your review

File: js/app.js (hunk 1 above)
Comment (Hamzaoutdoors) [reply js/app.js:152]: Done thanks for your review

File: README.md (hunk 5 above)
Comment (carlos-ssh) [reply README.md:1]: - External files are files that aren't needed into your project and there are persistent like this files hosted in your root directory: .DS_Store, .vscode. For this kind of files or directories are recommended to host in your .gitignore.
Screen Shot 2021-07-28 at 18 09 18

File: js/app.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque porro aliquam quod repellat harum distinctio quas voluptates eligendi maiores at facilis asperiores, voluptatem placeat cum nostrum nisi quae, natus neque? Minus expedita, aliquid repellat recusandae laboriosam natus culpa odit.',
+    liveLink: '#',
+    sourceLink: '#',
+  },
+
+];
+
+const divProject = document.createElement('div');
+
+function addEvents() {
```
Comment (rloterh) [inline js/app.js:100]: - [ ] OPTIONAL
 The `See Project` button when clicked does not display a popup with details about the project. Please see as specified in the project **Portfolio: details popup window** [documentation guide](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md). Kindly endeavour to improve upon this aspect, and other outlined specifications in this guide

File: js/app.js (hunk 1 above)
Comment (rloterh) [reply js/app.js]: Issue well resolved! ✔️

File: js/app.js (hunk 2 above)
Comment (rloterh) [reply js/app.js]: Issue well resolved! ✔️

File: js/app.js (hunk 4 above)
Comment (rloterh) [reply js/app.js]: Issue well resolved! ✔️

File: README.md (hunk 5 above)
Comment (rloterh) [reply README.md:1]: Issue not resolved! Kindly ensure to include the directory link of the `DS_Store` in the `.gitignore` file.

**Review by rloterh — APPROVED** [review]
Hi Hamza, 

Great job for making the necessary changes recommended by the previous reviewer. However, there's always room for improvement.
Kindly consider my OPTIONAL comments

## Optional suggestions
_Please keep in mind that you, for best practices and following the project requirements and the design templates closely kindly consider these aspects as outlined in the below comments._

**_Kindly notice that these are required changes that I'm sorry they didn't show up in the earlier review._**

<br>
Your project is complete! Congratulations! 🥂There is nothing else to say other than... it's time to merge it 👍.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**